presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoConnection.java

```diff
@@ -805,4 +797,23 @@ private static String getIsolationLevel(int level)
         }
         throw new SQLException("Invalid transaction isolation level: " + level);
     }
+
+    private String getSource()
```

Comment:
I find it a bit unintuitive that `getSource` does not simply return a `source` value. Maybe `Source` could get a default value that could evaluated using the below code?

Reply:
@kokosing thank you for the feedback!

`ConnectionProperty` does have a `getDefault` method but the issue I see with this is the logic to construct `source` depends on another connection property (`applicationNamePrefix`) and I don't see a way to access other connection properties from within a `ConnectionProperty` instance.

Is this what you were thinking? Is there a way to do this that I am missing?

Reply:
Ok, then let's rename the method. `getActualSource`?

Reply:
Sounds good. Updated.